language/move-vm/integration-tests/src/tests/loader_tests.rs

```diff
@@ -477,6 +475,34 @@ fn relink_load_err() {
     adapter.call_function_with_error(&b0, ident_str!("b0"));
 }
 
+#[test]
+fn relink_type_identity() {
+    let data_store = InMemoryStorage::new();
+    let mut adapter = Adapter::new(data_store);
+
+    let b0 = ModuleId::new(DEFAULT_ACCOUNT, ident_str!("b").to_owned());
+    let c0 = ModuleId::new(DEFAULT_ACCOUNT, ident_str!("c").to_owned());
+    let b1 = ModuleId::new(UPGRADE_ACCOUNT, ident_str!("b").to_owned());
+    let c1 = ModuleId::new(UPGRADE_ACCOUNT, ident_str!("c").to_owned());
+    let c0_modules = get_relinker_tests_modules_with_deps("c_v0", []).unwrap();
+    let c1_modules = get_relinker_tests_modules_with_deps("c_v1", []).unwrap();
+    let b1_modules = get_relinker_tests_modules_with_deps("b_v1", ["c_v1"]).unwrap();
+
+    adapter.publish_modules(c0_modules);
+    let c0_s = adapter.load_type(&TypeTag::from_str("0x2::c::S").unwrap());
+
+    let mut adapter = adapter.relink(UPGRADE_ACCOUNT, BTreeMap::from_iter([(b0, b1), (c0, c1)]));
+
+    adapter.publish_modules(c1_modules);
+    adapter.publish_modules(b1_modules);
+
+    let c1_s = adapter.load_type(&TypeTag::from_str("0x2::c::S").unwrap());
+    let b1_s = adapter.load_type(&TypeTag::from_str("0x2::b::S").unwrap());
+
+    assert_eq!(c0_s, c1_s);
+    assert_ne!(c1_s, b1_s);
```

Comment:
Phew! I didn't spot this `ne` at first and thought they were `eq` and spent much too much time staring at it trying to figure out why...

Reply:
All part of my master plan to make `amnn::coin::Coin == `sui::coin::Coin`.